### ISSUE TITLE
Cinnamon installs: Shutdown menu has no buttons other than cancel

### ISSUE THREAD

**Issue by un1versal** [linuxmint/Cinnamon#5444]
Adding Cinnamon ontop say Xubuntu or regular Ubuntu, kills the shutdown (quit) menu options.

#### System information 
![capture](https://cloud.githubusercontent.com/assets/3521959/16760348/2977259c-4813-11e6-9d74-e3269fb80caf.PNG)

#### How to duplicate
* Select "Quit" from the Cinnamon menu (which allows you to shutdown the computer) doesn't work properly - the dialog doesn't display any buttons other than "Cancel".

![capture](https://cloud.githubusercontent.com/assets/3521959/16759957/6c577684-4811-11e6-9fa5-3bf3cd35d02b.PNG)

#### There is luckily a workaround

Why doesnt this work by default? (hence the report)
```
gsettings set org.cinnamon.desktop.session settings-daemon-uses-logind true
gsettings set org.cinnamon.desktop.session session-manager-uses-logind true
gsettings set org.cinnamon.desktop.session screensaver-uses-logind false
```

#### After applying workaround and restarting Cinnamon.

![capture](https://cloud.githubusercontent.com/assets/3521959/16759813/ac2292fe-4810-11e6-98c4-2aef6e3d59da.PNG)

Im personally OK with the result of workaround (its better than nothing),  
In previous incarnations of Cinnamon, didnt this menu have some power menu on icon?

**Comment by JosephMcc** [comment]
It isn't default because until the recent Mint18 release Mint wasn't using systemd.

**Comment by un1versal** [comment]
yes but this is for ubuntu 16.04 base which does use systemd...

so I presume this is fixed now or whta does you closing this mean?

**Comment by mainmachine** [comment]
I would like to know if this has been fixed as well. The workaround works, but it seems like a serious error, and odd that it's so easy to fix from a (super)user standpoint...

Just did fresh installs recently on several of my machines to bring everything up to 16.04, so of course encountered this on every one.

**Comment by JosephMcc** [comment]
> I would like to know if this has been fixed as well. The workaround works, but it seems like a serious error, and odd that it's so easy to fix from a (super)user standpoint...

No it isn't "fixed" because it isn't actually broken. The defaults were chosen because they best suited installs on Linux Mint. That's why they are so easy to change. Because the defaults aren't the best choice everywhere. This is the sort of things I imagine would normally be modified by the maintainer of Cinnamon for a specific distro. As far as I know there is no proper maintainer for Ubuntu though I'm surprised it isn't modified at the Debian level. As far as I know we only get this complaint from Ubuntu users, which means Cinnamon maintainers for systemd based distros like Arch and Fedora handle it.

I would imagine the defaults will be changed at some point since Mint18 now uses systemd but LMDE2 still does not.

**Comment by mainmachine** [comment]
@JosephMcc - That's completely fair. I pull from a ppa so as to be on the latest stable version of Cinnamon, so I'm not actually using the version maintained, patched, etc. by Ubuntu proper - it's determined by the ppa maintainer.

I suspected the choice might have been to avoid conflict with those settings' defaults on other desktops, thank you for confirming that, and explaining it quite well. :)

**Comment by un1versal** [comment]
Well, I'll send an email off to the ppa maintainer (which is a 3rd party equivalent to AUR), there's no actual Ubuntu maintainers for Cinnamon 3.x afaik, and Ill link this report.

Then I *imagine* Ill chase this Rabbit for a while just out of curiosity to see how many holes I chase the rabbit down into. :/

**Comment by mainmachine** [comment]
@un1versal - here are the two ppa's I am aware of that server up 3.0.x:

https://launchpad.net/~kranich/+archive/ubuntu/cinnamon

https://launchpad.net/~embrosyn/+archive/ubuntu/cinnamon

I am currently running on the latter, as it is at 3.0.7, while Kranich's is still at 3.0.6...

Let me know if you do this or not, as if you don't get around to it I might...

**Comment by un1versal** [comment]
@mainmachine I use also https://launchpad.net/~embrosyn/+archive/ubuntu/cinnamon

I havent gotten around to it yet, I dont expect this week be any good either, problem is I may forget, so best to double up efforts anyway. 

What do you think?

**Comment by mainmachine** [comment]
@un1versal - just contacted Embrosyn on launchpad. 

I'll report back when he/she responds. :)

**Comment by un1versal** [comment]
@mainmachine I have contacted all ppas 3 ppa owners since they all have same issue, lets see what they say.

**Comment by embrosyn** [comment]
Hi everyone, 

embrosyn here. I have received mails from both @mainmachine and @un1versal. Sorry for the not-so-timely response. Treat this comment as a reply to both of your mails. It is certainly going to be long enough as if I wrote you both a proper reply :).

> This [defaults] is the sort of things I imagine would normally be modified by the maintainer of Cinnamon for a specific distro.

Hmm, do you mean the defaults located at "cinnamon-desktop/schemas/org.cinnamon.desktop.session.gschema.xml.in.in"? When I first encountered it myself, I found the [solution on AskUbuntu](http://askubuntu.com/questions/691813/cinnamon-desktop-clicking-menu-shutdown-presents-no-real-button-options) (which is also mentioned on my PPA) and wanted to make it permanent by altering the earlier mentioned file. The reason I wasn't bold enough to simply do this (even though it seems like a logical change) is that Debian maintainers (from which Ubuntu tends to import Cinnamon packages and who definitely know what they are doing more than I am) did a GSettings override (located at "debian/libcinnamon-desktop4.gsettings-override.linux") instead. As I want to do the least hacky, originally envisioned option, I would like to hear your informed opinion on which one of options is the one to go for.

> As far as I know there is no proper maintainer for Ubuntu ...

Unless if my English really sucks (and it really could, I'm not a native speaker), this seems kinda offensive. We're all trying to do our best and provide latest Cinnamon versions to Ubuntu users whilst not sacrificing functionality.

Cheers,
embrosyn

P.S. For further reference, I'm a "he" ;).

**Comment by un1versal** [comment]
@embrosyn ;) nice... Hi there! and welcome.

thx so much for reply, and thank you again for providing this ppa and its Cinnamon packages. As a current user, minus this *minor* issue this ticket is about, your efforts are indeed appreciated.

For future users only the implementation of solution would be great as things should work out of the box in this respect, but I dont really know much about it to make any sort of educated guess.

>...  I'm not a native speaker), this seems kinda offensive....

A lot of meaning is often lost between writer and reader, as a non-native English speaker myself I would not have inferred offense from that, as It *feels* like the meaning was of a official, stamped and endorsed by Canonical/Debian and shipped with Ubuntu and variants by default much like the 2/x Cinnamon is.

Luckily for us (me) users we (I) dont have to wait for Debian or Ubuntu politics and policies to catch up to enjoy Cinnamon 3.x and we (I) can indeed count our (my)  blessings since people like you actually go above and beyond to provide quality packaging,...

TL;DR

thx again I hope theres a fair and permanent fix possible for the issue at hand.

see you around.

**Comment by JosephMcc** [comment]
> Unless if my English really sucks (and it really could, I'm not a native speaker), this seems kinda offensive.

It's not meant to be. I'm referring to the official repos, not PPA maintainers. In the past the version of Cinnamon in the Ubuntu repos has been very outdated and if I remember correctly at one point the version of Cinnamon was even a development version and not a release version.

I would probably just go with the gsettings override. We do that for a few Cinnamon and Nemo settings even in Mint.

**Comment by un1versal** [comment]
After looking at **https://launchpad.net/ubuntu/+archive/primary/+files/cinnamon-desktop_3.0.2-2.debian.tar.xz**
 it seems like @embrosyn pointed out they use a gsettings override. heck the file is even in there ready to go.

**Comment by embrosyn** [comment]
@un1versal 
Thanks for the warm welcoming :). I'm glad people actually make use of my PPA in the sea of various (probably decent, but didn't test them personally) Cinnamon PPAs (ek0892, Tsvetko, Sunderland93...). That heavy disclamatory usage of pronouns in parentheses reminiscent of legalese documents though, it made me chuckle :). I think I have resolved the *minor* issue pointed out in this, er, issue *cinnamon-desktop*, version 3.0.2-1~xenial1 (Wily build coming soon) using GSettings override as @JosephMcc recommended. I have set up a VM with vanilla Ubuntu Xenial to try this out, and ... it works! See screenshots [1](http://i.imgur.com/mxBSjqb.png) and [2](http://i.imgur.com/Ov5e1qe.png) (ignore the unappealing black background).

@JosephMcc
Really, really sorry for the misunderstanding. PPAs are often treated as second-class citizens (untrusted source, packages built by God knows who, potentially malicious code, etc.) when compared to official distribution repositories, so I did an unintended prejudicial misinterpretation there. Sure, some PPAs *might* host malicious, broken code and whatnot, but I'm doing my best so that my PPA isn't one of them :).

And yeah, often largely outdated official Ubuntu Cinnamon builds are the primary reason why Cinnamon PPAs will probably never go out of fashion.

**Comment by mainmachine** [comment]
@embrosyn - thanks for the good work, and for addressing this issue!

I think gsettings overide is a perfectly good way to go about it, I'm more familiar with postinst scripts than with gsettings specifically, so that's why I suggested that as an approach.

I accept this as a substitue to a direct email about my questions. :+1:

**Comment by un1versal** [comment]
yep fantastic, now new users to @embrosyn ppa dont have to suffer the issue and have it fully working out-of-the-box.
 
@embrosyn I actually tested a few ppa's including nightly, they all have this issue, I also contacted all other ppa owners, but you were by far the quickest to reply and address this issue.

so from now on I can safely recommend your ppa as a primary goto for *stable* 3.x Cinnamon builds.

SO now lets see how long for all the others to catch up.

Thx again.

 @JosephMcc sorry for the noise caused and thanks for allowing us to use this ticket to sort through this.

**Comment by mainmachine** [comment]
Indeed, thanks to @JosephMcc for the thankless work you do. :)

**Comment by JosephMcc** [comment]
Thanks guys, but you did all the work and handled the problem.

**Comment by un1versal** [comment]
@JosephMcc ;) and you help make cinnamon in first place, so thank you to all the "spice" crew for making out desktop lives a little saner and more bearable.
